addon/components/challenges/add.js

```diff
@@ -54,10 +51,99 @@ export default Ember.Component.extend(ClickOutside, {
     saveDismiss(){
       this.actions.save.bind(this)();
       this.actions.dismiss.bind(this)();
+    },
+    selectMajor(){
+      if(this.get('new.minor')){
+        this.actions.switchBump.bind(this)();
+      }
+    },
+    selectMinor(){
+      if(this.get('new.major')){
+        this.actions.switchBump.bind(this)();
+      }
+    },
+    switchBump(){
+      this.toggleProperty('new.minor');
+      this.toggleProperty('new.major');
+    },
+    repoCreate(){
+      let regex = /https:\/\/github.com\/(\w+?\/[\w|-]+)\/?.*/;
+      let link = this.get('new.link').match(regex);
+
+      if(link){
+        let id = link[1];
+        this.get('store').findRecord('repo', id).then(repo => {
+          this.get('model.repos').addObject(repo);
+          this.get('model.resources').addObject({
+            route: repo.get('id'),
+            type: 'repo'
+          });
+          this.get('model').save().then(() => {
+            this.actions.cleanUp.bind(this)();
+          });
+        });
+      }
+      else{
+        return null;
+      }
+    },
+    milestoneCreate(){
+      if(!this.get('new.description')){
+        return null;
+      }
+      let version = semver.parse(this.get('model.challenges.lastObject.title'));
+      if(this.get('new.major')) {
+        version[0]++;
+      }
+      else if(this.get('new.minor')) {
+        version[1]++;
+      }
+      this.get('model.repos').forEach((repo) =>{
+        let milestone = this.get('store').createRecord('milestone',{
+          title: `${version[0]}.${version[1]}.0`,
+          description: this.get('new.description'),
+          state: 'open',
+          repo: repo
+        });
+        milestone.save().then(() => {
+          this.actions.cleanUp.bind(this)();
+        });
+      });
+    },
+    issueCreate(){
+      if( this.get('new.title') && this.get('new.description') && this.get('new.Milestone') && this.get('new.Repository') && this.get('new.Priority') && this.get('new.Type')){
+        let route = this.get('new.Repository');
+        this.get('store').findRecord('label',  route + '/labels/prio:' + this.get('new.Priority')).then((prioLabel) => {
+          this.get('store').findRecord('label',  route + '/labels/type:' + this.get('new.Type')).then((typeLabel) => {
+            let labels = [prioLabel, typeLabel];
+            let milestoneNr = this.get('model.challenges')
+            .findBy('title',this.get('new.Milestone'))
+            .milestones.findBy('repo.fullName', route)
+            .get('number');
+            let issue = this.get('store').createRecord('issue', {
+              title: this.get('new.title'),
+              body: this.get('new.description'),
+              milestone: milestoneNr,
+              labels: labels
```

Comment:
I tried to fix this, but it is a necessity to ensure that the labels are fetched before they become part of the "create"